docs/contributing/making-a-pull-request.md

```diff
@@ -0,0 +1,450 @@
+    the discussion respectful at all times.
+
+    It is important to note that not all pull requests get incorporated int the
+    codebase. The reasons can vary. The work may bring to light other issues that
+    block integration of the pull request. Sometimes it helps uncover better ways of
+    doing things or shows that a more general approach is needed. All of this is
+    fine and helps the project progress, even if specific changes are not,
+    ultimately, accepted.
+
+4. Make any requested changes by committing them to your local clone and
+   pushing them up to your fork. This will automatically update the pull request.
+   It may well take a few iterations to get your contributions to an acceptable
+   state. You can help the process along by carefully reading comments made and
+   making changes with care.
+
+5. Once the reviewer is fully satisfied with the changes, they can merge them
+   into the main branch (or 'master'). In the process, they may 'squash' your
+   commits together into a smaller number of commits and may edit the messages
+   that describe them. Congratulations, you have now contributed to this project
+   and should see the changes in the main branch under your name.
+
+6. You can now delete the fork and your local repository and start afresh again
+   next time around. Alternatively, you can keep the repository and local clone
+   around but it is important that you keep them in a good state for any
+   subsequent work. We recommend that you start by deleting the branch you used
+   on your fork.
+
+7. To make sure you have the changes you produced, pull them from the main
+   repository into the main branch of your fork.
+
+8. Similarly, delete the topic branch from your local clone and...
+
+9. pull the changes to its master branch.
+
+## Steps
+
+Now that the overall process is outlined, here are specific instructions and
+tips. There are many choices to be made when describing a process for
+contributing to a project via a pull request. In the following, we assume that
+you are working with the Git command-line tools. For most alternatives (such as
+using IDEs or using functionality provided through the GitHub web interface),
+the translation from the command-line instructions should be simple enough.  We
+will add notes only where really necessary to keep the complexity of this to a
+reasonable level.
+
+### Forking the repository
+
+To make changes to Material for MkDocs, you would first fork one of its
+repositories on GitHub. This is so that you have a repository on GitHub that
+you can push changes to (only maintainers and collaborators have write access
+to the original repositories).
+
+Fork the [repository for the public version] if you want to make changes to
+code that is in the public version or if you want to make changes to the
+documentation. Please make sure to change the name of the repository by
+appending `-fork` to that people who come across it know that they have found a
```

Comment:
In principle, I agree but the note is actually easy to miss in the heat of the battle. Not a hill I would die on but I have always hated running into forks and not realizing I was not dealing with the original, maintained version. Seen students run into this trap as well. Not a battle I would die on but I think this is a small thing to do that can prevent silly mistakes.

Reply:
Then I'd say maybe reformulate from an imperative into a suggestion. "A good idea is ... so that users don't start creating issues on your repository that are meant for us" or something more up your gusto.